astropy/uncertainty/core.py

```diff
@@ -268,3 +268,7 @@ def pdf_histogram(self, **kwargs):
         bin_edges = np.array(bin_edges)
         be_shape = self.shape + (bin_edges.size//self.size,)
         return nhists.reshape(nh_shape), bin_edges.reshape(be_shape)
+
+    def to(self, *args, **kwargs):
+        tod = self.distribution.to(*args, **kwargs)
+        return self.__class__(tod)
```

Comment:
I think this has to be `Distribution(tod)` - it will infer the right mixin class from `tod` (just like in `__array_ufunc__`).